specs/agents/tracing-instrumentation-aws-lambda.md

```diff
@@ -200,3 +200,7 @@ Field | Value | Description | Source
 Lambda functions are immediately frozen as soon as the handler method ends. In case APM data is sent in an asyncronous way (as most of the agents do by default) data can get lost if not sent before the lambda function ends.
 
 Therefore, the Lambda instrumentation has to ensure that data is flushed in a blocking way before the execution of the handler function ends.
+
+# Flushed Signal
+Some Lambda functions will use the custom-built Lambda extension that allows the agent to send its data locally. The extension asynchronously forwards the data it receives from the agent to the APM server so the Lambda function can return its result with minimal delay. In order for the extension to know when it can flush its data, it must receive a signal indicating that the lambda function has completed. There are two possible signals: one is via a subscription to the AWS Lambda Logs API and the other is an agent intake request with the query param `flushed=true`. A signal from the agent is preferrable because there is an inherent delay with the sending of the Logs API signal.
+Therefore, the agent should send its final intake request at the end of the function invocation with the query param `flushed=true`. In case there is no more data to send at the end of the function invocation, the agent should send an empty intake request with this query param.
```

Comment:
@elastic/apm-agent-python I remember discussing the empty intake request with the flushed query param with you because of how agent data is sent. I don't remember exactly why so can you tell me if the last sentence is relevant?

Reply:
We queue our data blind to the actual sending of the data. I think we can set a flag when we call the final flush, that will result in the `flushed=true` being set on the last payload. But it's technically a race condition so we'll also have the empty payload logic as a backup to make sure `flushed=true` does get sent.